test/e2e/index.ts

```diff
@@ -0,0 +1,145 @@
+	const clientKey = fs.readFileSync(mTLS.keyPath);
+
+	const agent = new https.Agent({
+		cert: clientCert,
+		key: clientKey,
+		rejectUnauthorized: false,
+	});
+
+	return (request, init) => fetch(request, { ...init, agent });
+}
+
+async function getIssuerConfig(name: string, mTLS?: MTLSConfiguration) {
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(`https://${name}${PRIVATE_TOKEN_ISSUER_DIRECTORY}`);
+	const config: IssuerConfig = (await response.json()) as IssuerConfig;
+
+	const token = config['token-keys'].find(
+		token => token['token-type'] == TOKEN_TYPES.BLIND_RSA.value
+	);
+
+	if (!token) {
+		throw new Error('Could not find BlindRSA token key on issuer');
+	}
+
+	const publicKeyEnc = base64UrlToUint8Array(token['token-key']);
+	const publicKey = await importPublicKey(util.convertRSASSAPSSToEnc(publicKeyEnc));
+
+	return {
+		publicKey,
+		publicKeyEnc,
+		url: `https://${name}${config['issuer-request-uri']}`,
+	};
+}
+
+async function importPublicKey(spki: Uint8Array) {
+	return crypto.subtle.importKey('spki', spki, { name: 'RSA-PSS', hash: 'SHA-384' }, true, [
+		'verify',
+	]);
+}
+
+async function test(issuerName: string, mTLS?: MTLSConfiguration) {
+	console.log(`
+████████╗███████╗███████╗████████╗   ███████╗██████╗ ███████╗
+╚══██╔══╝██╔════╝██╔════╝╚══██╔══╝██╗██╔════╝╚════██╗██╔════╝
+   ██║   █████╗  ███████╗   ██║   ╚═╝█████╗   █████╔╝█████╗  
+   ██║   ██╔══╝  ╚════██║   ██║   ██╗██╔══╝  ██╔═══╝ ██╔══╝  
+   ██║   ███████╗███████║   ██║   ╚═╝███████╗███████╗███████╗
+   ╚═╝   ╚══════╝╚══════╝   ╚═╝      ╚══════╝╚══════╝╚══════╝
+`);
+	console.log('Testing with issuer', issuerName);
+	const client = new Client(BlindRSAMode.PSS);
+
+	const redemptionContext = new Uint8Array(32);
+	redemptionContext.fill(0xfe);
+	const challenge = new TokenChallenge(TOKEN_TYPES.BLIND_RSA.value, issuerName, redemptionContext);
+
+	const {
+		publicKeyEnc: issuerPublicKeyEnc,
+		publicKey: issuerPublicKey,
+		url: issuerRequestURL,
+	} = await getIssuerConfig(issuerName);
+
+	const tokenRequest = await client.createTokenRequest(challenge, issuerPublicKeyEnc);
+
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(issuerRequestURL, {
+		method: 'POST',
+		headers: {
+			'Content-Type': MediaType.PRIVATE_TOKEN_REQUEST,
+			'Accept': MediaType.PRIVATE_TOKEN_RESPONSE,
+		},
+		body: Buffer.from(tokenRequest.serialize()),
+	});
+
+	if (!response.ok) {
+		throw new Error(`Issuer request failed: ${response.status} ${response.statusText}`);
+	}
+
+	const tokenResponse = publicVerif.TokenResponse.deserialize(
+		new Uint8Array(await response.arrayBuffer())
+	);
+	const token = await client.finalize(tokenResponse);
+
+	console.log(
+		`Issuer tokens are valid: ${
+			(await verifyToken(BlindRSAMode.PSS, token, issuerPublicKey)) &&
+			response.headers.get('Content-Type') === MediaType.PRIVATE_TOKEN_RESPONSE
+		}`
+	);
+}
+
+async function main() {
```

Comment:
the entire body of the test seems to me that can be wrapped with a regular Jest test.
No need of using a CLI, other that for passing params, which can be passed and stored in a `testdata` folder.

Reply:
I agree. At the same time, this repository is not tighten to any specific deployment.
I'll see how to update this repository to use wrangler.toml configuration in test:e2e target.